src/core/HyperlanePermissionlessDeployer.ts

```diff
@@ -105,88 +104,75 @@ export class HyperlanePermissionlessDeployer {
+    ismDeployer.cacheAddressesMap(
+      objMerge(sdkContractAddressesMap, addressesMap),
+    );
+    const ismFactoryContracts = await ismDeployer.deploy(this.chains);
+    addressesMap = this.writeMergedAddresses(addressesMap, ismFactoryContracts);
+    this.logger(`ISM factory deployment complete`);
+
+    // 2. Deploy IGPs to all chains
+    this.logger(`Deploying IGP contracts`);
+    const igpConfig = buildIgpConfigMap(owner, this.chains);
+    const igpDeployer = new HyperlaneIgpDeployer(this.multiProvider);
+    igpDeployer.cacheAddressesMap(addressesMap);
+    const igpContracts = await igpDeployer.deploy(igpConfig);
+    addressesMap = this.writeMergedAddresses(addressesMap, igpContracts);
+    this.logger(`IGP deployment complete`);
+
+    // 3. Deploy core contracts to local chain
+    this.logger(`Deploying core contracts to ${this.local}`);
+    // Build an IsmFactory that covers all chains so that we can
+    // use it later to deploy ISMs to remote chains.
+    const ismFactory = HyperlaneIsmFactory.fromAddressesMap(
+      objMerge(sdkContractAddressesMap, addressesMap),
+      this.multiProvider,
+    );
+    const coreDeployer = new HyperlaneCoreDeployer(
+      this.multiProvider,
+      ismFactory,
     );
+    coreDeployer.cacheAddressesMap(addressesMap);
+    const coreConfig = buildCoreConfigMap(owner, this.local, this.remotes);
+    const coreContracts = await coreDeployer.deploy(coreConfig);
+    addressesMap = this.writeMergedAddresses(addressesMap, coreContracts);
     this.logger(`Core deployment complete`);
-    addresses = this.writeMergedAddresses(addresses, coreContracts);
-
-    // Next, deploy MultisigIsms to the remote chains
-    // TODO: Would be cleaner if using HyperlaneIsmDeployer
-    const isms: ChainMap<MultisigIsmContracts> = {};
-    isms[this.local] = {
-      multisigIsm: coreContracts[this.local].multisigIsm,
-    };
-    for (const remote of this.remotes) {
-      this.logger(`Deploying multisig ISM to chain ${remote}`);
-      isms[remote] = {
-        multisigIsm: await coreDeployer.deployLegacyMultisigIsm(
-          remote,
-          coreConfig[remote].multisigIsm,
-        ),
-      };
-      this.logger(`Deployment of multisig ISM to chain ${remote} complete`);
-    }
-    this.logger(`ISM deployment complete`);
-    addresses = this.writeMergedAddresses(addresses, isms);
-
-    // Next, deploy TestRecipients to all chains
-    const testRecipientConfig: ChainMap<TestRecipientConfig> = objMap(
-      isms,
-      (chain, ism) => {
-        return { ism: ism.multisigIsm.address };
-      },
+
+    // 4. Deploy ISM contracts to remote chains
+    this.logger(`Deploying ISMs to ${this.remotes}`);
+    const ismConfig = buildIsmConfigMap(owner, this.remotes, this.chains);
+    const ismContracts = await promiseObjAll(
+      objMap(ismConfig, async (chain, config) => {
+        return {
+          interchainSecurityModule: await ismFactory.deploy(chain, config),
```

Comment:
Optional: This structure allows for some parallelism but we may get better logging + error handling if we do them sequentially (e.g. if one chain fails, is it clear from logs which one?)